lib/master/worker_entry.go

```diff
@@ -53,6 +54,8 @@ type workerEntry struct {
 	expireAt time.Time
 	state    workerEntryState
 
+	receivedFinish atomic.Bool
```

Comment:
Finished is not a state for worker? 
why we can't use `workEntryFinished`?

Reply:
`Finished` is technically a second component of the state. Formally, introducing the `receivedFinish` field makes the FSM states a Cartesian product `(workerEntryState, isFinished)`. For example, we can have `(workerEntryNormal, true)` or `(workerEntryOffline, false)`. `receivedFinish` is similar to a `signal-received` flag, which needs to be handled in parallel to its other states.